src/plugins/intel_cpu/src/emitters/plugin/aarch64/jit_emitter.cpp

```diff
@@ -98,136 +106,109 @@ void jit_emitter::emitter_preamble(const std::vector<size_t>& in_idxs,
+        aux_gpr_idxs.push_back(idx);
     }
 
+    const uint32_t end_gpr_idx = Xbyak_aarch64::Operand::X28;
+    for (size_t gpr_idx = 0; gpr_idx <= end_gpr_idx; ++gpr_idx) {
+        size_t _idx = end_gpr_idx - gpr_idx; // we allocate from the end
+
+        if (aux_gpr_idxs.size() >= get_aux_gprs_count()) break;
+        if (_idx == Xbyak_aarch64::Operand::X18) continue;
+
+        if (std::find(aux_gpr_idxs.begin(), aux_gpr_idxs.end(), _idx) != aux_gpr_idxs.end()) continue;
+        if (std::find(preserved_gpr_idxs.begin(), preserved_gpr_idxs.end(), _idx) != preserved_gpr_idxs.end()) continue;
+
+        aux_gpr_idxs.push_back(_idx);
+        preserved_gpr_idxs.push_back(_idx);
+    }
+    if (aux_gpr_idxs.size() < get_aux_gprs_count())
+        OV_CPU_JIT_EMITTER_THROW("Failed to allocate required number of general-purpose registers");
+
     if (!entry_map_.empty()) {
         // last aux_gpr_idx is for p_table, we can use aux_gpr_idxs from idx 0 for other purpose
         p_table = Xbyak_aarch64::XReg(aux_gpr_idxs[aux_gpr_idxs.size() - 1]);
         aux_gpr_idxs.erase(aux_gpr_idxs.end() - 1);
     }
 
-    for (size_t i = 0; i < preserved_gpr_idxs.size(); ++i) {
-        h->str(Xbyak_aarch64::XReg(preserved_gpr_idxs[i]), pre_ptr(h->sp, -16));
-    }
+    store_context(preserved_gpr_idxs, preserved_vec_idxs);
 
     if (!entry_map_.empty()) {
         load_table_addr();
     }
 }
 
 void jit_emitter::emitter_postamble() const {
-    const int size = static_cast<int>(preserved_gpr_idxs.size());
-    for (int i = (size - 1); i >= 0; --i) {
-        h->ldr(Xbyak_aarch64::XReg(preserved_gpr_idxs[i]), post_ptr(h->sp, 16));
-    }
+    restore_context(preserved_gpr_idxs, preserved_vec_idxs);
+
+    preserved_vec_idxs.clear();
     preserved_gpr_idxs.clear();
 
     aux_vec_idxs.clear();
     aux_gpr_idxs.clear();
 }
 
 void jit_emitter::store_context(const std::unordered_set<size_t>& ignore_registers) const {
-    // 1. General-purpose Registers
-    // 1.1. store pair registers
-    const auto store_gpr_regs_size = store_gpr_regs.size();
-    const auto last = store_gpr_regs_size % 2;
-    for (size_t i = 0; i < (store_gpr_regs_size - last); i += 2) {
-        h->stp(Xbyak_aarch64::XReg(store_gpr_regs[i]),
-               Xbyak_aarch64::XReg(store_gpr_regs[i + 1]),
-               pre_ptr(h->sp, -get_gpr_length() * 2));
-    }
+    store_context(store_gpr_regs, vec_regs, ignore_registers);
+}
 
-    // 1.1. store the remaining register
-    if (last != 0) {
-        h->str(Xbyak_aarch64::XReg(store_gpr_regs[store_gpr_regs_size - 1]),
-               pre_ptr(h->sp, -get_gpr_length() * 2));
+void jit_emitter::store_context(
+        const std::vector<size_t>& gpr_regs,
+        const std::vector<size_t>& vec_regs,
+        const std::unordered_set<size_t>& ignore_vec_regs) const {
+    for (const auto i : gpr_regs) {
+        h->str(Xbyak_aarch64::XReg(i), pre_ptr(h->sp, -get_gpr_length() * 2));
```

Comment:
I might miss something, but we do not need to multiply with 2, as we are using h->str not h->std. Same comment with counterpart in restore_context.

Reply:
fixed, thanks